ddf-rpc-api/src/main/java/com/connexta/ddf/catalog/direct/CatalogMethods.java

```diff
@@ -701,6 +800,62 @@ private Date parseDate(Serializable date) {
     return new Date(ZonedDateTime.from(temporalAccessor).toInstant().toEpochMilli());
   }
 
+  private ImmutablePair<TermFacetProperties, Error> getFacetTerm(
+      Map<String, Object> paramProperties) {
+    Map<String, Object> facets = (Map<String, Object>) paramProperties.get("facets");
+
+    if (!(facets.get("facetAttributes") instanceof Collection)
+        || ((Collection) facets.get("facetAttributes")).isEmpty()) {
+      return pairOf(
+          null,
+          rpc.error(
+              Error.INVALID_PARAMS,
+              "facetAttributes was not a Collection with at least one attribute",
+              mapOf(
+                  "path",
+                  asList("params", "properties", "facets", "facetAttributes"),
+                  "irritant",
+                  facets.get("facetAttributes"))));
+    }
+    Set<String> facetAttributes = new HashSet<>((Collection) facets.get("facetAttributes"));
+
+    SortFacetsBy sortKey = SortFacetsBy.COUNT;
+    if (facets.containsKey("sortKey") && facets.get("sortKey") instanceof String) {
+      String sortKeyString = (String) facets.get("sortKey");
+      if (sortKeyString.equalsIgnoreCase("INDEX")) {
+        sortKey = SortFacetsBy.INDEX;
+      } else if (sortKeyString.equalsIgnoreCase("COUNT")) {
+        sortKey = SortFacetsBy.COUNT;
+      } else {
+        return pairOf(
+            null,
+            rpc.error(
+                Error.INVALID_PARAMS,
+                "sortKey was not a String of `INDEX` or `COUNT`",
+                mapOf(
+                    "path",
+                    asList("params", "properties", "facets", "sortKey"),
+                    "irritant",
+                    facets.get("sortKey"))));
+      }
+    }
+
+    int facetLimit = 100;
+    if (facets.containsKey("facetLimit") && facets.get("facetLimit") instanceof Integer) {
```

Comment:
✏️ These two conditionals can also be simplified to just the `instanceof` check.